### PR TITLE
Small performance related changes.

### PR DESCRIPTION
**Pull request by bordoni**

### CODE REVIEW

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+/**
+ * Class Bar
+ *
+ * @since   5.0.0
+ *
+ * @package Tribe\Events\Views\V2\Admin
+ */
+class Bar {
+	CONST SUSPEND_CACHE_KEY = 'tribe_events_suspend_view_html_cache';
+	CONST SUSPEND_NONCE_KEY = 'tribe_events_views_v2_suspend_view_html_cache';
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: * :no_entry_sign: PHP keywords must be lowercase; expected "const" but found "CONST"
* :no_entry_sign: CONST keyword must be lowercase; expected "const" but found "CONST"

<sub>hash: 435cbfd526d829082f83521957b14805</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: * :no_entry_sign: PHP keywords must be lowercase; expected "const" but found "CONST"
* :no_entry_sign: CONST keyword must be lowercase; expected "const" but found "CONST"

<sub>hash: d8ba633cfaf311a169524172b9e9b2f0</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+	 * @return  boolean  If the current user has suspended the cache for views.
+	 */
+	public function is_view_html_cache_suspended() {
+		// Check anything if you have WP_DEBUG turned on.
+		if ( ! defined( 'WP_DEBUG' ) ) {
+			return false;
+		}
+
+		// You can only suspend when you can `manage_options`
+		if ( ! current_user_can( 'manage_options' ) ) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 21935f25ebcee516a7d1168c3dc27b24</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	 * @return void
+	 */
+	public function process_suspending_cache() {
+		// Check anything if you have WP_DEBUG turned on.
+		if ( ! defined( 'WP_DEBUG' ) ) {
+			return;
+		}
+
+		$actions_allowed = [
+			'unsuspend_view_cache' => 0,
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 46ac062d5fa579fafa7db0e862976aac</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+			'suspend_view_cache' => 1,
+		];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :warning: Array double arrow not aligned correctly; expected 3 space(s) between "'suspend_view_cache'" and double arrow, but found 1.

<sub>hash: ce0c7c8e006e19796bac08f5074de464</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+			return;
+		}
+
+		if ( ! wp_verify_nonce( tribe_get_request_var( '_wpnonce' ), static::SUSPEND_NONCE_KEY ) ) {
+			return;
+		}
+
+		update_user_option( get_current_user_id(), static::SUSPEND_CACHE_KEY, $actions_allowed[ $action ] );
+		wp_safe_redirect( remove_query_arg( [ 'action', '_wpnonce' ] ) );
+		exit;
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 5a204cfd94a536fddac8b18c1b94d29a</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+			return;
+		}
+
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		$is_suspended = $this->is_view_html_cache_suspended();
+		$action = $is_suspended ? 'unsuspend_view_cache' : 'suspend_view_cache';
+		$label  = $is_suspended ? '<span style="color: red;">' . __( 'Unsuspend View Cache', 'the-events-calendar' ) . '</span>' : __( 'Suspend View Cache', 'the-events-calendar' );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :warning: Equals sign not aligned with surrounding assignments; expected 7 spaces but found 1 space

<sub>hash: afa7969f8688b57c469dd79918b92d56</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :warning: Equals sign not aligned with surrounding assignments; expected 8 spaces but found 2 spaces

<sub>hash: e3781ccafb596410ace44c081476c8bb</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+		$args = [
+			'id'     => 'tribe-events-views-v2-suspen_view_html_cache',
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: b4dfff724448e5ed2d2c28d734d63e44</sub>

File: src/Tribe/Views/V2/Admin/Bar.php [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+			'title'  => $label,
+			'parent' => 'top-secondary',
+			'href'   => wp_nonce_url( add_query_arg( [ 'action' => $action ] ), static::SUSPEND_NONCE_KEY )
+		];
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Admin/Bar.php]: * :no_entry_sign: Each line in an array declaration must end in a comma
* :no_entry_sign: Short array syntax is not allowed
* :no_entry_sign: Each array item in a multi-line array declaration must end in a comma

<sub>hash: 1e0babacfe30b11c4f4c1fc46117d237</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 11]
```diff
@@ -60,6 +61,8 @@ protected function add_actions() {
 		add_action( 'tribe_events_pro_shortcode_tribe_events_after_assets', [ $this, 'action_disable_shortcode_assets_v1' ] );
 		add_action( 'updated_option', [ $this, 'action_save_wplang' ], 10, 3 );
 		add_action( 'the_post', [ $this, 'manage_sensitive_info' ] );
+		add_action( 'admin_bar_menu', [ $this, 'action_include_admin_bar_actions' ] );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: a758b7bfc313826f8e8acabaff18da8c</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 12]
```diff
@@ -60,6 +61,8 @@ protected function add_actions() {
...
+		add_action( 'parse_request', [ $this, 'action_process_suspending_cache_from_admin_bar' ] );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 982e215cf5431c237bbd1d2533b7f5b1</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 13]
```diff
@@ -88,6 +91,7 @@ protected function add_filters() {
 		add_filter( 'tribe_general_settings_tab_fields', [ $this, 'filter_general_settings_tab_live_update' ], 20 );
 		add_filter( 'tribe_events_rewrite_i18n_slugs_raw', [ $this, 'filter_rewrite_i18n_slugs_raw' ], 50, 2 );
 		add_filter( 'tribe_get_event_after', [ $this, 'filter_events_properties' ] );
+		add_filter( 'tribe_events_views_v2_should_cache_html', [ $this, 'filter_suspend_cache_from_admin_bar' ] );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: d1a0453174cefbcfc5288234ff0b3ea8</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 14]
```diff
@@ -138,6 +142,15 @@ public function action_disable_shortcode_assets_v1() {
 		$assets->disable_v1();
 	}
 
+	/**
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Doc comment for parameter "$wp_admin_bar" missing

<sub>hash: 60a1db76fa11a72a49ded6985582b236</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 15]
```diff
@@ -162,7 +175,7 @@ public function on_wp_head() {
 	 *
 	 * @since 4.9.2
 	 *
-	 * @param  \Tribe__Events__Rewrite  $rewrite  An instance of the Tribe rewrite abstraction.
+	 * @param  TEC_Rewrite  $rewrite  An instance of the Tribe rewrite abstraction.
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Expected 1 spaces after parameter type; 2 found

<sub>hash: 2aa0048970cd3885df8fcd84859eefb2</sub>

File: src/Tribe/Views/V2/Hooks.php [hunk 16]
```diff
@@ -458,12 +471,29 @@ public function filter_general_settings_tab_live_update( $fields ) {
 		return $fields;
 	}
 
+	/**
+	 * Ignore all other rules suspend html views cache for this user.
+	 *
+	 * @since  5.0.0
+	 *
+	 * @param  boolean  $should_cache  Weather or not caching was previously active.
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Hooks.php]: :no_entry_sign: Expected 1 spaces after parameter type; 2 found

<sub>hash: 536e2096667a6bd58ea1fbc4df7266b8</sub>

**Comment by borkweb** [comment]
Unless these are bugfixes, let’s PR these for release 5.0.1

**Comment by bordoni** [comment]
Yeah for sure, they are meant for 5.0.1, my mistake.

**Pull request closed without merging**